server.go

```diff
@@ -247,20 +258,27 @@ func handlePacket(s *Server, p interface{}) error {
 	case *sshFxInitPacket:
 		return s.sendPacket(sshFxVersionPacket{sftpProtocolVersion, nil})
 	case *sshFxpStatPacket:
-
-		// stat the requested file
-		info, err := s.driver.Stat(p.Path)
-		if err != nil {
-			return s.sendError(p, err)
-		}
-		return s.sendPacket(sshFxpStatResponse{
-			ID:   p.ID,
-			info: info,
-		})
 	case *sshFxpLstatPacket:
 		// stat the requested file
 		info, err := s.driver.Stat(p.Path)
 		if err != nil {
+			if err != os.ErrNotExist {
+				return s.sendError(p, err)
+			}
+			// if the file doesn't exist, see if we have an open handle for that file, and send info based on that
```

Comment:
can we include in this comment that we do this to better mirror OpenSSH behavior?